examples/step-81/step-81.cc

```diff
@@ -0,0 +1,894 @@
+    SparsityPattern           sparsity_pattern;
+    SparseMatrix<double>      system_matrix;
+    Vector<double>            solution;
+    Vector<double>            system_rhs;
+  };
+
+  // @sect3{Class Template Definitions and Implementation}
+  //
+  // @sect4{The Constructor}
+  // The Constructor simply consists of default initialization a number of
+  // discretization parameters (such as the domain size, mesh refinement,
+  // and the order of finite elements and quadrature) and declaring a
+  // corresponding entry via ParameterAcceptor::add_parameter(). All of
+  // these can be modified by editing the .prm file. Absorbing boundary
+  // conditions can be controlled with the absorbing_boundary boolean. If
+  // absorbing boundary conditions are disabled we simply enforce
+  // homogeneous Dirichlet conditions on the tangential component of the
+  // electric field. In the context of time-harmonic Maxwell's equations
+  // these are also known as perfectly conducting boundary conditions.
+
+  template <int dim>
+  Maxwell<dim>::Maxwell()
+    : ParameterAcceptor("Maxwell")
+    , dof_handler(triangulation)
+  {
+    ParameterAcceptor::parse_parameters_call_back.connect(
+      std::bind(&Maxwell<dim>::parse_parameters_callback, this));
+
+    scaling = 20;
+    add_parameter("scaling", scaling, "scale of the hypercube geometry");
+
+    refinements = 8;
+    add_parameter("refinements",
+                  refinements,
+                  "number of refinements of the geometry");
+
+    fe_order = 0;
+    add_parameter("fe order", fe_order, "order of the finite element space");
+
+    quadrature_order = 1;
+    add_parameter("quadrature order",
+                  quadrature_order,
+                  "order of the quadrature");
+
+    absorbing_boundary = true;
+    add_parameter("absorbing boundary condition",
+                  absorbing_boundary,
+                  "use absorbing boundary conditions?");
+  }
+
+
+  template <int dim>
+  void Maxwell<dim>::parse_parameters_callback()
+  {
+    fe = std::make_unique<FESystem<dim>>(FE_NedelecSZ<dim>(fe_order), 2);
+  }
+
+  // The Maxwell::make_grid() routine creates the mesh for the
+  // computational domain which in our case is a scaled square domain.
+  // Additionally, a material interface is introduced by setting the
+  // material id of the upper half ($y>0$) to 1 and of the lower half
+  // ($y<0$) of the computational domain to 2.
+  // We are using a block decomposition into real and imaginary matrices
+  // for the solution matrices. More details on this are available
+  // under the Results section.
+
+  template <int dim>
+  void Maxwell<dim>::make_grid()
+  {
+    GridGenerator::hyper_cube(triangulation, -scaling, scaling);
+    triangulation.refine_global(refinements);
+
+    if (!absorbing_boundary)
+      {
+        for (auto &face : triangulation.active_face_iterators())
+          if (face->at_boundary())
+            face->set_boundary_id(1);
+      };
```

Comment:
```suggestion
      }
```
The same at many other places.